public/app/features/dashboard/state/DashboardMigrator.ts

```diff
@@ -777,6 +777,18 @@ export class DashboardMigrator {
       }
     }
 
+    if (oldVersion < 37) {
+      panelUpgrades.push((panel: PanelModel) => {
+        if (panel.options?.legend && panel.options.legend.displayMode === 'hidden') {
+          panel.options.legend.displayMode! = 'list';
+          panel.options.legend.showLegend! = false;
+        } else {
+          panel.options.legend.showLegend! = true;
+        }
+        return panel;
+      });
+    }
```

Comment:
This migration is fine, but I think you haven't bumped the dashboard schema version on line 79 - hence the migration is not kicking in.

Please also add a test for this new migration.